src/test/java/uk/gov/pay/products/persistence/dao/ProductDaoTest.java

```diff
@@ -67,4 +67,33 @@ public void shouldReturnProductsWithStatusActive() throws Exception {
         assertThat(products.get(0).getExternalId(), is(externalId));
         assertThat(products.get(0).getStatus(), is(ProductStatus.ACTIVE));
     }
+
+    @Test
+    public void shouldUpdateBatchServiceName() throws Exception {
+        String externalId = randomUuid();
```

Comment:
I would probably add 3 products for this test. Including 1 of a different gatewayAccountId. And check that has not changed in the end. That kind of proves at affects only for a given GAID